File: src/main/java/es/udc/fi/dc/fd/model/services/StockMarketService.java
package es.udc.fi.dc.fd.model.services;

import es.udc.fi.dc.fd.model.common.exceptions.*;
import es.udc.fi.dc.fd.model.entities.Enterprise;
import es.udc.fi.dc.fd.model.entities.User;
import es.udc.fi.dc.fd.model.entities.OrderLine.*;
import es.udc.fi.dc.fd.model.services.exceptions.*;
import es.udc.fi.dc.fd.rest.dtos.AnnualBenefitsListDto;
import java.time.LocalDateTime;

public interface StockMarketService {

	public Enterprise createEnterprise(Long userId, Enterprise enterprise)
			throws DuplicateInstanceException, PermissionException, NumberException;

	public Enterprise createAnnualBenefits(Long userId, Long enterpriseId, AnnualBenefitsListDto benefitsList)
			throws DuplicateInstanceException, PermissionException, InstanceNotFoundException, InvalidArgumentException;

	public float transfer(Long userId, Float money, String operation)
			throws InvalidOperationException, InstanceNotFoundException, NotEnoughBalanceException;

	public long order(Long owner, OrderType orderType, OrderLineType orderLineType, Float price, int number, Long enterpriseId, LocalDateTime deadline)
			throws NotEnoughBalanceException, NotOwnedException, NotAvaliableException;
	
	public void deleteOrder(Long owner, Long orderId, Boolean avaliable)
			throws NotOwnedException, InstanceNotFoundException, NotAvaliableException;

	public Enterprise modifyAvaliableEnterprise(Long creator, Long enterpriseId, Boolean avaliable)
			throws NotCreatorException, InstanceNotFoundException;
	
	public int searchUserActionsNumber(User user, Enterprise enterprise, Boolean sellOnlyNotAbaliable);
}

File: frontend/src/backend/stockMarketService.js
import {
	fetchConfig,
	appFetch,
} from "./appFetch";


export const transfer = (money, operation, onSuccess,onErrors) =>{
	appFetch('/market/transfer', fetchConfig('POST', { money, operation }), onSuccess, onErrors);
	}

export const createEnterprise = (enterprise, onSuccess, onErrors) =>
	appFetch('/market/create_enterprise', fetchConfig('POST', enterprise), onSuccess, onErrors);

export const updateEnterprise = (id,annualBenefits,onSuccess, onErrors) =>
	appFetch(`/market/update_enterprise/${id}`, fetchConfig('PUT', annualBenefits), onSuccess, onErrors);
	
export const createOrder = (order, onSuccess, onErrors) =>
	appFetch(`/market/order`, fetchConfig('POST', order), onSuccess, onErrors);

